src/server.cpp

```diff
@@ -2313,6 +2313,28 @@ void Server::SendBlocks(float dtime)
 	m_clients.unlock();
 }
 
+bool Server::SendBlock(session_t peer_id, v3s16 blockpos)
+{
+	MapBlock *block = nullptr;
+	try {
+		block = m_env->getMap().getBlockNoCreate(blockpos);
+	} catch (InvalidPositionException &e) {};
+	if (!block)
+		return false;
+
+	m_clients.lock();
```

Comment:
Can you please write more than 3 words here? I do not understand "it's not solved". Please explain how "it's not solved" so I can attempt to understand what that statement means.

Reply:
replace lock/unlock with std::lock_guard. As lock & unlock functions are defined the lock_guard can replace manual lock/unlock. Please look at other code parts using that

Reply:
```
$ git pull --rebase origin master
X11 forwarding request failed on channel 0
remote: Enumerating objects: 29, done.
remote: Counting objects: 100% (29/29), done.
remote: Total 35 (delta 28), reused 29 (delta 28), pack-reused 6
Unpacking objects: 100% (35/35), done.
From github.com:minetest/minetest
 * branch              master     -> FETCH_HEAD
   e22a69d6..eadcbe47  master     -> origin/master
First, rewinding head to replay your work on top of it...
Applying: Force send a mapblock to a player.
Applying: Always send mapblock on teleport or respawn.
$ grep -ri guard src
src/debug.cpp:	case EXCEPTION_GUARD_PAGE:
src/debug.cpp:		return "Attempted guard page access";
$
```

Reply:
Invalid request. Please implement this first in a few places so that it becomes established code style and locking protection. After that, we can use it as a requirement for new commits to adhere to this newly established coding standard.

Until this is an established coding standard, the request is invalid.

Reply:
Do we use some other form of RAII locks elsewhere?

Reply:
@rubenwardy yes we use them in some places

Reply:
As far as I can see, [lock_guard is not used anywhere in minetest](https://github.com/minetest/minetest/search?q=lock_guard&unscoped_q=lock_guard), but it's done like this:
https://github.com/minetest/minetest/blob/b298b0339c79db7f5b3873e73ff9ea0130f05a8a/src/util/container.h#L86

Maybe add a function
```cpp
MutexAutoLock& lock() { return MutexAutoLock(m_clients_mutex); }
```
to `ClientInterface`?